nshlib/nsh_proccmds.c

```diff
@@ -613,29 +614,54 @@ int cmd_ps(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
 #ifndef CONFIG_NSH_DISABLE_KILL
 int cmd_kill(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
 {
+/* kill will send SIGTERM to the task in case no signal is selected by
+ * -<signal> option
+ */
+
   char *ptr;
   char *endptr;
   long signal;
   long pid;
 
-  /* Check incoming parameters.  The first parameter should be "-<signal>" */
-
-  ptr = argv[1];
-  if (*ptr != '-' || ptr[1] < '0' || ptr[1] > '9')
+  if (argc == 3)  /* kill -<signal> <pid> */
     {
-      goto invalid_arg;
-    }
+      /* Check incoming parameters.
+       * The first parameter should be "-<signal>"
+       */
 
-  /* Extract the signal number */
+      ptr = argv[1];
+      if (*ptr != '-' || ptr[1] < '0' || ptr[1] > '9')
```

Comment:
In the future to be more compatible with Unix and Linux, it should also accept the signal name, ie:
$ kill -SIGTERM 4297

Reply:
Yes, it will be a good improvement for a next PR.

Reply:
Nice! Please squash these 4 commits in a single commit.